docs/src/main/asciidoc/security-jdbc-guide.adoc

```diff
@@ -0,0 +1,318 @@
+----
+
+In our context, we are using H2 in-memory mode as identity store and we load `import.sql` to init the database wit users and roles.
+
+[source,sql]
+----
+CREATE TABLE test_user (
+  id INT,
+  username VARCHAR(255),
+  password VARCHAR(255),
+  role VARCHAR(255)
+);
+
+INSERT INTO test_user (id, username, password, role) VALUES (1, 'admin', 'admin', 'admin');
+INSERT INTO test_user (id, username, password, role) VALUES (2, 'user','user', 'user');
+----
+
+[NOTE]
+====
+We're sure you know but we kindly remind that you must not store clear password in production environment ;-)
+The `elytron-security-jdbc` offers a built-in bcrypt password mapper.
+====
+
+We can now configure the Elytron Jdbc Realm.
+
+[source,properties]
+----
+quarkus.security.jdbc.enabled=true
+quarkus.security.jdbc.principal-query.sql=SELECT u.password, u.role FROM test_user u WHERE u.username=? <1>
+quarkus.security.jdbc.principal-query.clear-password-mapper.enabled=true <2>
+quarkus.security.jdbc.principal-query.clear-password-mapper.password-index=1
+quarkus.security.jdbc.principal-query.attribute-mappings.0.index=2 <3>
+quarkus.security.jdbc.principal-query.attribute-mappings.0.to=groups
+----
+
+The `elytron-security-jdbc` extension requires at least one principal query to trust and load a user identity.
+
+<1> We define a SQL parametrized statement (with exactly 1 parameter) which should return the user's password plus any information your want to load.
+<2> We  configure the password mapper with the position of the password field in the `SELECT` fields and other information like salt, hash encoding, etc.
+<3> We use `attribute-mappings` to bind the `SELECT` projection fields (ie. `u.role` here) to the target Principal representation attributes.
+
+[NOTE]
+====
+In the `principal-query` configuration all the `index` properties start at 1 (rather than 0).
+====
+
+== Testing the Application
+
+The application is now protected and the identities is provided by our database.
+The very first todo is to ensure the anonymous access works.
+
+```bash
+$ curl -i -X GET http://localhost:8080/api/public
+HTTP/1.1 200 OK
+Content-Length: 6
+Content-Type: text/plain;charset=UTF-8
+
+public%
+```
+
+Now, let's try a to hit a protected resources anonymously.
+
+```bash
+$ curl -i -X GET http://localhost:8080/api/admin
+HTTP/1.1 401 Unauthorized
+Content-Length: 14
+Content-Type: text/html;charset=UTF-8
+
+Not authorized%
+```
+
+So far, so good,  now let's try with an allowed user.
+
+```bash
+$ curl -i -X GET -u admin:admin http://localhost:8080/api/admin
+HTTP/1.1 200 OK
+Content-Length: 5
+Content-Type: text/plain;charset=UTF-8
+
+admin%
+```
+By providing the `admin:admin` credentials, the extension succeed to authenticate the user and load its roles.
+The `admin` user is authorized to access to the protected resources.
+
+The  user the `admin` should be forbid to access a resources protected with `@RolesAllowed("user")` because it hasn't this role.

```

Comment:
```suggestion
The user `admin` should be forbidden to access a resource protected with `@RolesAllowed("user")` because it doesn't have this role.
```